pkg/services/sso/redhatsso_service.go

```diff
@@ -0,0 +1,199 @@
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared"
+	"github.com/golang/glog"
+	serviceaccountsclient "github.com/redhat-developer/app-services-sdk-go/serviceaccounts/apiv1internal/client"
+	"time"
+)
+
+var _ keycloakServiceInternal = &redhatssoService{}
+
+//////////////////////////////////////////////////////
+// Builder
+var _ KeycloakServiceBuilder = &redhatSSOServiceBuilder{}
+var _ RedhatSSOConfigurator = &redhatSSOConfigurator{}
+
+type redhatssoService struct {
+	client redhatsso.SSOClient
+}
+
+type RedhatSSOConfigurator interface {
+	WithRedhatSSOClient(client *redhatsso.SSOClient) KeycloakServiceBuilder
+	WithRedhatSSOConfiguration(conf *redhatsso.RedhatSSOConfig) KeycloakServiceBuilder
+}
+
+type redhatSSOServiceBuilder struct {
+	client *redhatsso.SSOClient
+}
+
+func (b redhatSSOServiceBuilder) Build() KeycloakService {
+	return &keycloakServiceProxy{
+		accessTokenProvider: *b.client,
+		service: &redhatssoService{
+			client: *b.client,
+		},
+	}
+}
+
+type redhatSSOConfigurator struct {
+}
+
+func (c redhatSSOConfigurator) WithRedhatSSOClient(client *redhatsso.SSOClient) KeycloakServiceBuilder {
+	return redhatSSOServiceBuilder{
+		client: client,
+	}
+}
+
+func (c redhatSSOConfigurator) WithRedhatSSOConfiguration(conf *redhatsso.RedhatSSOConfig) KeycloakServiceBuilder {
+	client := redhatsso.NewSSOClient(conf)
+	return redhatSSOServiceBuilder{
+		client: &client,
+	}
+}
+
+// END Builder
+//////////////////////////////////////////////////////
+
+func (r *redhatssoService) RegisterKafkaClientInSSO(accessToken string, kafkaNamespace string, orgId string) (string, *errors.ServiceError) {
+	svcData, err := r.client.CreateServiceAccount(accessToken, kafkaNamespace, fmt.Sprintf("%s:%s", orgId, kafkaNamespace))
+	if err != nil {
+		return "", errors.NewWithCause(errors.ErrorFailedToCreateSSOClient, err, "failed to register Kafka Client in SSO")
+	}
+	if svcData.Secret != nil {
+		return *svcData.Secret, nil
+	} else {
+		return "", errors.NewWithCause(errors.ErrorGeneral, err, "failed to retrieve secret while registering Kafka Client in SSO")
+	}
+}
+
+func (r *redhatssoService) RegisterOSDClusterClientInSSO(accessToken string, clusterId string, clusterOathCallbackURI string) (string, *errors.ServiceError) {
+	// TODO
+	return "", errors.New(errors.ErrorGeneral, "RegisterOSDClusterClientInSSO Not implemented")
+}
+
+func (r *redhatssoService) DeRegisterClientInSSO(accessToken string, clientId string) *errors.ServiceError {
+	err := r.client.DeleteServiceAccount(accessToken, clientId)
+	if err != nil {
+		return errors.NewWithCause(errors.ErrorFailedToDeleteSSOClient, err, "failed to delete the sso client")
+	}
+	glog.V(5).Infof("Kafka Client %s deleted successfully", clientId)
+	return nil
+}
+
+func (r *redhatssoService) GetConfig() *keycloak.KeycloakConfig {
+	return nil
+}
+
+func (r *redhatssoService) GetRealmConfig() *keycloak.KeycloakRealmConfig {
+	return nil
```

Comment:
Can we return the sso realm here? same mas-sso service

Reply:
That will be done in another PR (KeycloakRealmConfig must be changed to AuthRealmConfig and we will use one config for both services implementations)